### ISSUE TITLE
Gym jsbsim environment is not deterministic

### ISSUE THREAD

**Issue by fteicht** [galleon/gym-jsbsim#45]
I don't know if it is a bug or if it is intended but the taxiing environment (at least) is not deterministic. If it is intended, I think it should be clearly stated in the README since deterministic algorithms will fail to solve the problem.

Below is a simple code that shows that resetting the environment and stepping once with the same action produces different simulator states and sometimes different environment states too. The environment state is not systematically random when the simulator is because different simulator states can correspond to the same (abstract or partial) environment state.

Thus, if you confirm that it is fully normal that the simulator is non-deterministic, and knowing that the environment state is a part of the simulator state, it means that we are solving a Partially Observable Markov Decision Process. If so, please state it in the README since only POMDP or RL algorithms can handle it.

Code:

````python
import gym
import gym_jsbsim

env = gym.make('GymJsbsim-TaxiapControlTask-v0')

# For 5 samples actions
for i in range(5):
    a = env.action_space.sample()
    print('ACTION', str(a))
    reset_state = None
    state = None
    full_state = None
    # Reset and step once 5 times for the same action
    for j in range(5):
        rs = env.reset()
        s = env.step(a)
        fs = env.get_state()
        if state is None:
            reset_state = rs
            state = s
            full_state = fs
        else:
            print('TEST', str(j), ':',
                  str(reset_state == rs),
                  str(state == s),
                  str(full_state == fs))
            reset_state = rs
            state = s
            full_state = fs
```

Possible output:

```
ACTION (array([0.28284418]),)
TEST 1 : True True False
TEST 2 : True True False
TEST 3 : True True False
TEST 4 : True True True
ACTION (array([0.8231234]),)
TEST 1 : True True False
TEST 2 : True True True
TEST 3 : True True False
TEST 4 : True True False
ACTION (array([-0.60835098]),)
TEST 1 : True True False
TEST 2 : True False False
TEST 3 : True False False
TEST 4 : True True False
ACTION (array([-0.94829114]),)
TEST 1 : True False False
TEST 2 : True False False
TEST 3 : True True False
TEST 4 : True True False
ACTION (array([0.58901612]),)
TEST 1 : True True False
TEST 2 : True True False
TEST 3 : True True False
TEST 4 : True True False
```

**Comment by garibarba** [comment]
The environment state seems to be deterministic for the `HeadingControlTask`, but the simulator state is not. At least the landing gear friction coefficients seem to be to blame.

```python
import gym
import gym_jsbsim

env = gym.make('GymJsbsim-HeadingControlTask-v0')

# For 5 samples actions
for i in range(5):
    a = env.action_space.sample()
    print('ACTION', str(a))
    reset_state = None
    state = None
    full_state = None
    # Reset and step once 5 times for the same action
    for j in range(5):
        rs = env.reset()
        s = env.step(a)
        fs = env.get_state()
        if state is None:
            reset_state = rs
            state = s
            full_state = fs
        else:
            print('TEST', str(j), ':',
                  str(reset_state == rs),
                  str(state == s),
                  str(full_state == fs),
                  [p.name_jsbsim for p in fs if not full_state[p] == fs[p]]
                  )
            reset_state = rs
            state = s
            full_state = fs
```

Possible output

```python
ACTION (array([0.27265066]), array([0.03697818]), array([0.76023615]), array([0.45904392]))
TEST 1 : True True False ['gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 2 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit[1]/side_friction_coeff']
TEST 3 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit[1]/side_friction_coeff']
TEST 4 : True True False ['simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
ACTION (array([-0.31992279]), array([0.22968446]), array([0.32873589]), array([0.59068755]))
TEST 1 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit[1]/side_friction_coeff']
TEST 2 : True True False ['simulation/terminate', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 3 : True True False ['simulation/terminate', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 4 : True True False ['simulation/terminate', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff']
ACTION (array([0.7993829]), array([0.24780035]), array([-0.16169555]), array([0.81289914]))
TEST 1 : True True False ['gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 2 : True True False ['gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 3 : True True False ['simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 4 : True True False ['simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
ACTION (array([0.69921031]), array([0.08634777]), array([-0.06505904]), array([0.52719235]))
TEST 1 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 2 : True True False ['gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff']
TEST 3 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 4 : True True False ['gear/unit/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
ACTION (array([-0.92263752]), array([-0.61611608]), array([0.18940282]), array([0.01758615]))
TEST 1 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 2 : True True False ['simulation/terminate', 'simulation/pause', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
TEST 3 : True True False ['simulation/terminate', 'gear/unit/side_friction_coeff']
TEST 4 : True True False ['simulation/terminate', 'gear/unit/side_friction_coeff', 'gear/unit[1]/side_friction_coeff', 'gear/unit[2]/side_friction_coeff']
```

**Comment by bgaudron** [comment]
I had also observed different trajectories with the same sequence of action for the taxi environments.
This was caused by the line `propulsion_set_running: -1` in the initial_conditions of the taxi environments that initialized the engines before calling run_ic causing randomly the engines to stay off. 
This line has just been removed last friday in commit e8250f540311acb472a95cdd54f6e089baa011da of master.

@fteicht @garibarba I have rerun your tests and it looks like the only properties that are random from one simulation to another are the side friction coefficients only for the aircraft structures that are not in contact with the ground which should not be a problem.
